components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/introspection/OAuth2IntrospectionEndpoint.java

```diff
@@ -159,8 +161,14 @@ public Response introspect(@FormParam("token") String token, @FormParam("token_t
         // Provide token binding related info which required to validate the token.
         if (StringUtils.isNotBlank(introspectionResponse.getBindingType()) && StringUtils
                 .isNotBlank(introspectionResponse.getBindingReference())) {
-            respBuilder.setBindingType(introspectionResponse.getBindingType());
+            String bindingType = introspectionResponse.getBindingType();
+            respBuilder.setBindingType(bindingType);
             respBuilder.setBindingReference(introspectionResponse.getBindingReference());
+            if (OAuth2Constants.TokenBinderType.CERTIFICATE_BASED_TOKEN_BINDER.equals(bindingType) &&
+                    StringUtils.isNotBlank(introspectionResponse.getCnfBindingValue()) &&
+                    Boolean.parseBoolean(IdentityUtil.getProperty(OAuthConstants.ENABLE_TLS_CERT_TOKEN_BINDING))) {
```

Comment:
i think we can ommit following check here as we anyway check whether CERTIFICATE_BASED_TOKEN_BINDER token binder is used.

 Boolean.parseBoolean(IdentityUtil.getProperty(OAuthConstants.ENABLE_TLS_CERT_TOKEN_BINDING))

Reply:
Fixed in 1189e9d4d13c525be1351555f1270bf00635f164